test/python/circuit/test_circuit_qasm.py

```diff
@@ -768,6 +768,39 @@ def test_opaque_output(self):
 """
         self.assertEqual(qasm, expected)
 
+    def test_sequencial_inner_gates_with_same_name(self):
+        """Test if inner gates sequentially added with the same name result in the correct qasm"""
+        qubits_range = range(3)
+
+        gate_a = QuantumCircuit(3, name="a")
+        gate_a.h(qubits_range)
+        gate_a = gate_a.to_instruction()
+
+        gate_b = QuantumCircuit(3, name="a")
+        gate_b.append(gate_a, qubits_range)
+        gate_b.x(qubits_range)
+        gate_b = gate_b.to_instruction()
+
+        qc = QuantumCircuit(3)
+        qc.append(gate_b, qubits_range)
+        qc.z(qubits_range)
+
+        gate_a_id = id(qc.data[0].operation)
+
+        expected_output = """OPENQASM 2.0;
+include "qelib1.inc";
+gate a q0,q1,q2 {{ h q0; h q1; h q2; }}
+gate a_{0} q0,q1,q2 {{ a q0,q1,q2; x q0; x q1; x q2; }}
+qreg q[3];
+a_{0} q[0],q[1],q[2];
+z q[0];
+z q[1];
+z q[2];
+""".format(
+            gate_a_id
+        )
```

Comment:
Very minor, but it'll be a little easier to read this if you use f-string formatting.  It basically looks identical, except you have the opening quote look like `f"""` (e.g. `expected_output = f"""OPENQASM 2.0;`), in place of ``{0}`` on line 795 you can just write `{gate_a_id}`, and you won't need the `.format()` call.